probe-rs/src/architecture/xtensa/sequences/esp32s3.rs

```diff
@@ -6,17 +6,30 @@ use probe_rs_target::Chip;
 
 use super::XtensaDebugSequence;
 use crate::{
-    architecture::xtensa::communication_interface::XtensaCommunicationInterface, MemoryInterface,
+    architecture::{
+        esp_common::EspFlashSizeDetector,
+        xtensa::communication_interface::XtensaCommunicationInterface,
+    },
+    MemoryInterface,
 };
 
 /// The debug sequence implementation for the ESP32-S3.
 #[derive(Debug)]
-pub struct ESP32S3 {}
+pub struct ESP32S3 {
+    inner: EspFlashSizeDetector,
+}
 
 impl ESP32S3 {
     /// Creates a new debug sequence handle for the ESP32-S3.
     pub fn create(_chip: &Chip) -> Arc<dyn XtensaDebugSequence> {
-        Arc::new(Self {})
+        Arc::new(Self {
+            inner: EspFlashSizeDetector {
+                stack_pointer: 0x3FCF_0000,
```

Comment:
The intent for these to be temporary, though this information hardly ever needs to change - if I managed to find a working set of values for ESP32 and S2.